pkg/checksec/canary.go

```diff
@@ -19,6 +19,50 @@ type CanaryResult struct {
 // StackChk to check for stack_chk_fail value
 const StackChk = "__stack_chk_fail"
 
+// validatePath ensures the path is within the current working directory
+func validatePath(path string) error {
+	if path == "" {
+		return fmt.Errorf("path is empty")
+	}
+
+	// Disallow absolute paths
+	if filepath.IsAbs(path) {
```

Comment:
Why would an absolute path be an issue?

Reply:
I have encountered such security risks during previous operation and maintenance